test/memkind_memtier_data_movement_test.cpp

```diff
@@ -59,3 +60,34 @@ TEST_F(MemkindMemtierDataMovementTest,
     m_tier_memory = memtier_builder_construct_memtier_memory(m_builder);
     ASSERT_EQ(nullptr, m_tier_memory);
 }
+
+TEST(MemkindMemtierDataMovementBgThreadTest, test_bg_thread_lifecycle)
+{
+    struct memtier_memory *m_tier_memory = nullptr;
+    ProcStat proc_stat;
+
+    struct memtier_builder *m_builder =
+        memtier_builder_new(MEMTIER_POLICY_DATA_MOVEMENT);
+    ASSERT_NE(nullptr, m_builder);
+    int res = memtier_builder_add_tier(m_builder, MEMKIND_DEFAULT, 1);
```

Comment:
For now, we do not support building with tiers & static-ratio like policies. We need different parameters (such as DRAM_SIZE/ DRAM_RESERVATION_SIZE or sth similar).

This is not in the scope of this task. I think we could integrate the new interface with builders if the POC is successful. Please just create an instance of MTTAllocator and handle creating thread from there,